lib/datasource/metadata.ts

```diff
@@ -207,9 +206,9 @@ export function addMetaData(
   });
   extraBaseUrls.push('gitlab.com');
   if (dep.sourceUrl) {
-    const parsedUrl = URL.parse(dep.sourceUrl);
+    const parsedUrl = parseUrlLegacy(dep.sourceUrl);
```

Comment:
This works locally for me
```suggestion
    const parsedUrl = parseUrl(dep.sourceUrl);
```

Reply:
I tried that before and had a lot of failed datasource tests. Maybe because of invalid urls in tests.
Do not make this pr bigger i've decided to fix that in a later pr.